### DISCUSSION TITLE
reall really simple tasmota scripts

### DISCUSSION THREAD

**Discussion by julessilve** [tasmota/Berry_playground#37] [Tutorials and Code Blocks]
Hi Berry playground. 
I have spent half a day reading through the Berry Introduction, Berry Short Manual, Berry Cookbook and some of the posts/articles on this group. 
All of which leave me a bit bewildered. 
I am literally trying to switch a fan on and off at a certain temperature:
If Sensor:DS18x20 >50C then make GPIO2=1 else make GPIO2=0
end
As you can see I am no programmer but I am sure you get the drift of my intention. 
Is there a page with really really simple examples of Berry for Tasmota like this?
The other exaple I am trying to achieve is:
If GPIO4=1 then make GPIO2=1 else wait 3 hours then make GPIO2=0
end
Any advice is really appreciated. 
cheers...jules (rainy southern France)

**Comment by dkebler** [comment]
If you logic demands are simple, which they are, you might want to use Rules instead of Berry.  

so you can ask your question here instead https://github.com/arendst/Tasmota/discussions/categories/q-a

but I can get you started

read this.  https://tasmota.github.io/docs/Rules/

You need a couple trigger rules like this.  one for turn fan on above 50 and off below 48.  The two degree difference is some hysteresis  so the fan is not being turned on and off a lot near the set point.

`ON <trigger> DO <command> [ENDON | BREAK]`

your `<trigger>` is your 18x20 temperature probe sensor above or below some value and  if gpio2 is set in the config for relay1 then `power1` is what you want to set on and off in the `<command>` part.

Easy way to see exactly what those are is to look at the "regular" console/log.  Set the `teleperiod  10` in the console.  Then see what the "id" of the temperature is.  Click the `toggle` button and you should see in the log power1 being changed.

So what you'll need to do is figure out the correct syntax.  I leave that to you.  Find yourself some good examples that's how I figured it out.

btw your second question is also possible in rules.  your trigger is gpio4 (maybe power2 ?) .   Also using backlog you can set a timer to then trigger in 3 hours..  Bit more complicated but possible.

**Reply by julessilve** [reply]
ahhh many thanks. i was under the misunderstanding Berry replaced rules. i will have a read up on your suggestions.

thanks for the pointers.

cheers...j

**Reply by dkebler** [reply]
AFAIK rules will continue to be available for esp32.  Berry just takes advantage of esp32 greater processing and memory to extend the tasmota firmware with drivers and other dedicated control logic without having to compile into firmware binary.  In my case with berry I can make "self contained" complicated devices that don't depend on another machine  (i.e. network) elsewhere to function.

**Comment by dkebler** [comment]
watch this.

https://youtu.be/08_GBROKQH0?t=840&feature=shared

**Reply by julessilve** [reply]
brill. thanks a mill
